Framework/Algorithms/src/WorkspaceJoiners.cpp

```diff
@@ -56,19 +56,14 @@ WorkspaceJoiners::execWS2D(API::MatrixWorkspace_const_sptr ws1,
     auto &outSpec = output->getSpectrum(i);
     const auto &inSpec = ws1->getSpectrum(i);
 
-    // Copy X,Y,E
-    outSpec.setX(XValues);
-    outSpec.dataY() = inSpec.dataY();
-    outSpec.dataE() = inSpec.dataE();
+    outSpec.setHistogram(inSpec.histogram());
```

Comment:
I think these changes could give a substantial speedup for all algorithms that are based on `WorkspaceJoiners`. Could you tests that? A quick way to do that if you don't want to blindly implement a performance test is to just run the algorithms for a big enough workspace (using the doctest scripts can be a good starting point, if you adjust the workspace sizes).